packages/smooth_app/lib/pages/product/common/product_list_popup_items.dart

```diff
@@ -216,6 +225,135 @@ class ProductListPopupOpenInWeb extends ProductListPopupItem {
+      AppLocalizations.of(context),
+    );
+    final String fileName = _buildFileName(listName);
+    final StringBuffer csv = StringBuffer(
+        'Barcode${ProductListPopupItem.EXPORT_SEPARATOR}Name${ProductListPopupItem.EXPORT_SEPARATOR}Brand\n');
+
+    for (final String barcode in productList.barcodes) {
+      final Product? product = await DaoProduct(localDatabase).get(barcode);
+      if (product != null) {
+        final String name = (product.productName ?? '')
+            .replaceAll(ProductListPopupItem.EXPORT_SEPARATOR, '');
+        final String brand = (product.brands ?? '')
+            .replaceAll(ProductListPopupItem.EXPORT_SEPARATOR, '');
+        csv.write(
+            '"$barcode"${ProductListPopupItem.EXPORT_SEPARATOR}"$name"${ProductListPopupItem.EXPORT_SEPARATOR}"$brand"\n');
+      }
+    }
+
+    unawaited(
+      Share.shareXFiles(
+        <XFile>[
+          XFile.fromData(
+            utf8.encode(csv.toString()),
+            name: '$fileName.csv',
+            mimeType: 'text/csv',
+          ),
+        ],
+        fileNameOverrides: <String>['$fileName.csv'],
+      ),
+    );
+
+    return null;
+  }
+
+  String _buildFileName(String listName) {
+    final String name = listName.replaceAll(' ', '-').toLowerCase();
+    final String timestamp =
+        DateTime.now().toIso8601String().replaceAll(':', '_').split('.').first;
+
+    return '$name-$timestamp.csv';
+  }
+}
+
+/// Popup menu item for the product list page: import list.
+class ProductListPopupImport extends ProductListPopupItem {
+  @override
+  String getTitle(final AppLocalizations appLocalizations) =>
+      appLocalizations.product_list_import;
+
+  @override
+  IconData getIconData() => Icons.upload;
+
+  @override
+  ProductListPopupMenuEntry getEntry() => ProductListPopupMenuEntry.import;
+
+  @override
+  bool isDestructive() => false;
+
+  @override
+  Future<ProductList?> doSomething({
+    required final ProductList productList,
+    required final LocalDatabase localDatabase,
+    required final BuildContext context,
+  }) async {
+    final FilePickerResult? result = await FilePicker.platform.pickFiles(
+      type: FileType.custom,
+      allowedExtensions: <String>['csv'],
+    );
+
+    if (result == null) {
+      return null;
+    }
+
+    final File file = File(result.files.single.path!);
+    final String content = file.readAsStringSync();
+
+    final List<String> lines = content.split('\n');
+    final List<String> barcodes = <String>[];
+
+    for (final String line in lines) {
+      final List<String> parts = line.split(';');
+      if (parts.length < 3) {
+        continue;
+      }
+      final String barcode = parts[0].replaceAll('"', '');
+      barcodes.add(barcode);
+    }
+
+    final List<String> existingBarcodes =
+        await ProductRefresher().silentFetchAndRefreshListWithFeedback(
+              barcodes: barcodes,
+              localDatabase: localDatabase,
+              productType: ProductType.food,
```

Comment:
Only food?

Reply:
I've added a TODO here as I'm not sure how this should be handled. Do we have a universal function to fetch any type of product? Is there a way to know in advance just from the barcode? Should we add the information when exporting?